testsuite/cluster-test/src/effects/generate_cpu_flamegraph.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use crate::{effects::Action, instance::Instance};
+use anyhow::Result;
+use futures::future::{BoxFuture, FutureExt};
+use slog_scope::info;
+use std::fmt;
+
+pub struct GenerateCpuFlamegraph {
+    instance: Instance,
+    duration_secs: usize,
+}
+
+impl GenerateCpuFlamegraph {
+    pub fn new(instance: Instance, duration_secs: usize) -> Self {
+        Self {
+            instance,
+            duration_secs,
+        }
+    }
+}
+
+impl Action for GenerateCpuFlamegraph {
+    fn apply(&self) -> BoxFuture<Result<()>> {
+        async move {
+            info!("GenerateCpuFlamegraph {}", self.instance);
+            let cmd = format!(r#"
```

Comment:
Do we need some sort of `set -e` to make it fail if any command fails?

Reply:
Yes, that'll be a good idea